### PR TITLE
feat:CS-679 add pipelines for container image build, sonar scan, and trivy scan

### PR DESCRIPTION
**Pull request by niteshsakhiya**
## Description
Add pipelines for container image build, sonar scan, and trivy scan

## Why
To automate the container image build, scan the source code for sonar cloud analysis, and scan the container image using trivy

### CODE REVIEW

File: .github/workflows/build-and-publish-image-acr.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      - name: Azure CLI auth
+        uses: azure/login@v2
+        with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      # Login into Azure Container Registry (ACR)
+      - name: ACR Login
+        run: az acr login --name  ${{ secrets. AZURE_CONTAINER_REGISTRY }}
```
Comment (bmunguli) [inline .github/workflows/build-and-publish-image-acr.yaml:73]: These two steps here for ACR login and authentication i think are redunda and probably the reason why cofinity-x-sldt-bpn-discovery pipeline is failing on dependabot PRs, we need to change there too.  
The acr interaction happens again later in line 87

File: .github/workflows/build-and-publish-image-acr.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
```
Comment (bmunguli) [inline .github/workflows/build-and-publish-image-acr.yaml]: can we pin this to the latest version as a good practice 0.19.0: https://github.com/aquasecurity/trivy-action/tags

File: .github/workflows/build-test-scan-application.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Set up codeql
+        uses: github/codeql-action/init@v3
+        with:
+          languages: csharp
+
+      # Autobuild attempts to build any compiled languages (C/C++, C#, Go, Java, or Swift).
+      - name: Autobuild
```
Comment (bmunguli) [inline .github/workflows/build-test-scan-application.yaml]: can we do the build here with the dotnet building tools?

File: .github/workflows/build-and-publish-image-acr.yaml (hunk 1 above)
Comment (niteshsakhiya) [reply .github/workflows/build-and-publish-image-acr.yaml:73]: Fixed. Thanks for noticing. 
I have referred the pipeline from bpn discovery component. 😄

File: .github/workflows/build-test-scan-application.yaml (hunk 3 above)
Comment (niteshsakhiya) [reply .github/workflows/build-test-scan-application.yaml]: Done. Thanks.

File: .github/workflows/build-and-publish-image-acr.yaml (hunk 2 above)
Comment (niteshsakhiya) [reply .github/workflows/build-and-publish-image-acr.yaml]: Sure. It's done.

**Review by bmunguli — APPROVED** [review]

**Pull request merged**